lib/matplotlib/figure.py

```diff
@@ -122,7 +124,7 @@ def add(self, key, a):
         try:
             hash(key)
         except TypeError:
-            raise ValueError("first argument, %s, is not a valid key" % key)
```

Comment:
But if the same key comes in again it will get a different object which sort of defeats the point of this.